docs/advanced-features/athena.md

```diff
@@ -1,91 +1,475 @@
+
+ARN_ATHENA = f"arn:aws:s3:::{ATHENA_BUCKET}"
+ARN_CATALOG = f"arn:aws:glue:{REGION}:{ACCOUNT_ID}:catalog"
+#arn:aws:athena:us-east-1:712023778557:datacatalog/*
+ARN_DATACATALOG = f"arn:aws:glue:{REGION}:{ACCOUNT_ID}:datacatalog/*"
+ARN_DATABASE = f"arn:aws:glue:{REGION}:{ACCOUNT_ID}:database/{ATHENA_DB}"
+ARN_TABLE = f"arn:aws:glue:{REGION}:{ACCOUNT_ID}:table/{ATHENA_DB}"
+ARN_QUILT = f"arn:aws:s3:::{QUILT_BUCKET}"
+ARN_WORKGROUP = f"arn:aws:athena:{REGION}:{ACCOUNT_ID}:workgroup/{ATHENA_WORKGROUP}"
+
+# Create bucket in default region
+
+bucket = (
+    S3.create_bucket(Bucket=ATHENA_BUCKET)
+    if REGION == "us-east-1"
+    else S3.create_bucket(Bucket=ATHENA_BUCKET, CreateBucketConfiguration=location)
+)
+stat(bucket, ATHENA_BUCKET)
+# print(bucket)
+
+# Create Workgroup which outputs to that Bucket (if needed)
+
+wg_list = ATHENA.list_work_groups()
+wg_names = [wg["Name"] for wg in wg_list["WorkGroups"]]
+
+if ATHENA_WORKGROUP not in wg_names:
+    wg_create = ATHENA.create_work_group(
+        Name=ATHENA_WORKGROUP, Description="Quilt uses this for Athena SQL Queries"
+    )
+    stat(wg_create)
+
+# Configure Workgroup to use that Bucket
+wg_update = ATHENA.update_work_group(
+    WorkGroup=ATHENA_WORKGROUP,
+    ConfigurationUpdates={
+        "ResultConfigurationUpdates": {
+            "OutputLocation": ATHENA_URL,
+        },
+    },
+)
+stat(wg_update, "wg_update")
+
+# Create new GLUE Database
+
+db_create = ATHENA.start_query_execution(
+    QueryString=f"create database {ATHENA_DB}",
+    ResultConfiguration={"OutputLocation": ATHENA_URL + "/queries/"},
+)
+stat(db_create, "db_create")
+print(ATHENA_URL)
+print(ARN_DATABASE)
+```
+
+    mycompany-quilt-query-results: 200
+    wg_update: 200
+    db_create: 200
+    s3://mycompany-quilt-query-results
+    arn:aws:glue:us-east-1:712023778557:database/quilt_query
+
+
+## II. Granting Access to Athena
+
+By default, Quilt runs with very conservative permissions that do not allow access to [Amazon Athena](https://docs.aws.amazon.com/athena/latest/ug/what-is.html). To enable Athena SQL queries by your Quilt users, you must:
+
+### A. Create a new Athena policy.
+
+The standard [AmazonAthenaFullAccess](https://console.aws.amazon.com/iam/home#/policies/arn:aws:iam::aws:policy/AmazonAthenaFullAccess) policy is more permissive than necessary.  For production usage, we recommend creating a policy limited to only the above Bucket:
+<!--pytest-codeblocks:cont-->
+
+
+```python
+# https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/iam.html#IAM.ServiceResource.create_policy
+# https://docs.aws.amazon.com/athena/latest/ug/workgroups-access.html
+
+AthenaQuiltAccess = {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "GrantQuiltAthenaFullAccess",
+            "Effect": "Allow",
+            "Action": [
+                "athena:ListWorkGroups",
+            ],
+            "Resource": [
+                "*",
+            ],
```

Comment:
I'd write it like this for brevity, but that's up to you.
```suggestion
            "Action": "athena:ListWorkGroups",
            "Resource": "*",
```

Reply:
Alas, that is how Black does it.

Reply:
I'm pretty sure it doesn't add square brackets.